README.md

```diff
@@ -1,16 +1,532 @@
+particularly through [multi-order coverage (MOC)] maps and
+[hierarchical progressive surveys (HiPS)] used extensively in the [Aladin]
+astronomical information system. It is also used by LIGO and Virgo to store and
+communicate [gravitational-wave probability sky maps].
+
+HEALPix can be thought of as a tree. At the lowest resolution, level 0, HEALPix
+subdivides the sphere into 12 equal-area base tiles, assigned integer indicies
+0 through 11. At level 1, each of the 12 base tiles is subdivided into 4 tiles.
+Every subsequent level divides each of the preceding level's tiles into 4 new
+tiles. At a given level, each of the base pixels has been divided into
+4<sup>_level_</sup> pixels (_nside_ = 2<sup>_level_</sup> pixels on each side).
+Thus there are _npix_ = 12×4<sup>_level_</sup> pixels at a given resolution,
+assigned integer indices from 0 through (_npix_-1). This is called the NESTED
+indexing scheme. (There is also a RING indexing scheme in which the indicies
+advance from east to west and then from north to south.)
+
+A HEALPix tile, a node in the HEALPix tree, is fully addressed by three pieces
+of information: the indexing scheme (RING or NESTED), the resolution level
+(_level_ or equivalently _nside_), and the pixel index (_ipix_, an integer
+between 0 and _npix_-1).
+
+The image below, reproduced from https://healpix.jpl.nasa.gov, illustrates the
+first 4 levels of refinement of a HEALPix grid.
+
+![The first four levels of HEALPix subdivision of the unit sphere](https://healpix.jpl.nasa.gov/images/healpixGridRefinement.jpg)
+
+### HEALPix interval sets
+
+A region on the sphere can be encoded by a collection of disjoint HEALPix
+tiles, potentially at a mix of different resolution levels. Typically, large
+low-resolution tiles are used on the interior of the region, and small
+high-resolution tiles are used on the boundary. This is called a
+[multi-order coverage (MOC)] map. An example, reproduced from the
+[MOCPy documentation], is shown below.
+
+![An example HEALPix multi-order coverage map](https://cds-astro.github.io/mocpy/_images/plot_SDSS_r.png)
+
+Much like MOCs, LIGO/Virgo/KAGRA gravitational-wave probability sky maps are
+stored as multi-resolution HEALPix data sets, but with a vector of
+floating-point values attached to each tile. A multi-order refinement mesh from
+an example sky map is shown below, reproduced from [Singer & Price (2015)].
+
+![Example multi-order refinement mesh from a gravitational-wave probability sky map](https://emfollow.docs.ligo.org/userguide/_images/healpix-adaptive-mesh.svg)
+
+[Reinecke & Hivon (2015)][multi-order] introduced HEALPix interval sets as an
+alternative encoding of MOCs that enables fast and simple unions,
+intersections, and queries. In an interval set, each HEALPix tile is described
+by the interval of pixel indices at some very high resolution
+_level_<sub>max</sub> that are descendents of that tile. In an interval set, a
```

Comment:
So, just to confirm, descendent pixels are always labeled consecutively.

Reply:
Yes.